src/riak_kv_qry_compiler.erl

```diff
@@ -58,19 +58,30 @@
     {ok, [?SQL_SELECT{}]} | {error, any()}.
 compile(?DDL{}, ?SQL_SELECT{is_executable = true}, _MaxSubQueries) ->
     {error, 'query is already compiled'};
-compile(?DDL{} = DDL,
-        ?SQL_SELECT{is_executable = false, 'SELECT' = Sel} = Q, MaxSubQueries) ->
-    if Sel#riak_sel_clause_v1.clause == [] ->
-            {error, 'full table scan not implemented'};
-        el/=se ->
-            case compile_select_clause(DDL, Q) of
-                {ok, S} ->
-                    compile_where_clause(DDL, Q?SQL_SELECT{'SELECT' = S}, MaxSubQueries);
-                {error, _} = Error ->
-                    Error
-            end
+compile(?DDL{ table = T} = DDL,
+        ?SQL_SELECT{is_executable = false} = Q1, MaxSubQueries) ->
+    Mod = riak_ql_ddl:make_module_name(T),
+    case maybe_compile_group_by(Mod, compile_select_clause(DDL, Q1), Q1) of
```

Comment:
assing through the error from maybe_compile_group_by is clunky - either a double case or a single case like
```
    case {compile_select, compile_group_by, compile_where} of
      	{ok, S}, {ok, G}, {ok, W} -> record
      	{E1, E2, E2}              -> make_error(E1, E2, E3)
    end
```